ts/util/accountManager.ts

```diff
@@ -236,3 +244,153 @@ export async function registrationDone(ourPubkey: string, displayName: string) {
   // this will make the poller start fetching messages
   trigger('registration_done');
 }
+
+export const deleteDbLocally = async () => {
+  window?.log?.info('last message sent successfully. Deleting everything');
+  await window.persistStore?.purge();
+  window?.log?.info('store purged');
+
+  await deleteAllLogs();
+  window?.log?.info('deleteAllLogs: done');
+
+  await Data.removeAll();
+  window?.log?.info('Data.removeAll: done');
+
+  await Data.close();
+  window?.log?.info('Data.close: done');
+  await Data.removeDB();
+  window?.log?.info('Data.removeDB: done');
+
+  await Data.removeOtherData();
+  window?.log?.info('Data.removeOtherData: done');
+
+  window.localStorage.setItem('restart-reason', 'delete-account');
+};
+
+export async function sendConfigMessageAndDeleteEverything() {
+  try {
+    // DELETE LOCAL DATA ONLY, NOTHING ON NETWORK
+    window?.log?.info('DeleteAccount => Sending a last SyncConfiguration');
+
+    // be sure to wait for the message being effectively sent. Otherwise we won't be able to encrypt it for our devices !
+    await forceSyncConfigurationNowIfNeeded(true);
+    window?.log?.info('Last configuration message sent!');
+    await deleteDbLocally();
+    window.restart();
+  } catch (error) {
+    // if an error happened, it's not related to the delete everything on network logic as this is handled above.
+    // this could be a last sync configuration message not being sent.
+    // in all case, we delete everything, and restart
+    window?.log?.error(
+      'Something went wrong deleting all data:',
+      error && error.stack ? error.stack : error
+    );
+    try {
+      await deleteDbLocally();
+    } catch (e) {
+      window?.log?.error(e);
+    } finally {
+      window.restart();
+    }
```

Comment:
this could actually be made a finally on the outer try/catch, but it's probably easier to read like this